### PR TITLE
Add posts to project and add post view test

### LINKED ISSUE #130: Add has_many posts to project
`Project` right now is missing the `has_many :posts`. We need to add to both the model and the view.

For #26.

### PR DESCRIPTION
**Pull request by joshsmith**
Closes #130.

Simply adds posts to projects, expands on post some and tests it.

### CODE REVIEW

**Comment by joshsmith** [comment]
Ready for review.

File: web/controllers/post_controller.ex [hunk 1]
```diff
@@ -7,15 +7,18 @@ defmodule CodeCorps.PostController do
...
 
   def index(conn, %{"project_id" => project_id}) do
-    posts = Repo.all(
-      from c in Post,
-      where: c.project_id == ^project_id,
-      select: c
-    )
+    posts =
+      Post
+      |> Post.for_project(project_id)
```
Comment (npendery) [inline web/controllers/post_controller.ex]: I would move this to the model helpers we have been adding as a project filter

File: web/controllers/post_controller.ex [hunk 2]
```diff
@@ -36,22 +41,28 @@ defmodule CodeCorps.PostController do
...
-    post = Repo.one(
-      from c in Post,
-      where: c.number == ^number,
-      where: c.project_id == ^project_id,
-      select: c
-    )
+    post =
+      Post
+      |> Post.for_project(project_id)
+      |> Post.with_number(number)
```
Comment (npendery) [inline web/controllers/post_controller.ex]: include this as `show_filters` and add `with_number` to model helpers

File: web/controllers/post_controller.ex (hunk 1 above)
Comment (joshsmith) [reply web/controllers/post_controller.ex]: Which ones?

Comment (npendery) [reply web/controllers/post_controller.ex]: [These from my PR](https://github.com/code-corps/code-corps-phoenix/pull/139/files#diff-a8651bfb80ad2539a4f87ff71b4aa05cR16)

**Comment by npendery** [comment]
@JoshSmith the PR with the `ModelHelper` additions is in if you want to add `with_number` and `for_project` to that list

File: test/views/post_view_test.exs [hunk 3]
```diff
@@ -0,0 +1,64 @@
+defmodule CodeCorps.PostViewTest do
+  use CodeCorps.ConnCase, async: true
+
+  alias CodeCorps.Repo
+
+  # Bring render/3 and render_to_string/3 for testing custom views
+  import Phoenix.View
```
Comment (HeroicEric) [inline test/views/post_view_test.exs]: might be a good idea to only import the functions needed

``` elixir
import Phoenix.View, only: [render: 3, render_to_string: 3]
```

**Comment by joshsmith** [comment]
Could use a review on my latest commit.

File: web/models/model_helpers.ex [hunk 4]
```diff
@@ -15,6 +15,7 @@ defmodule CodeCorps.ModelHelpers do
 
   def id_filter(query, %{"filter" => %{"id" => id_list}}) do
     ids = id_list |> coalesce_id_string
+    IO.inspect ids
```
Comment (noslouch) [inline web/models/model_helpers.ex]: probably don't want this log statement here

Comment (joshsmith) [reply web/models/model_helpers.ex]: What if I do?

Comment (noslouch) [reply web/models/model_helpers.ex]: then whomever is watching `stdout` in production will see lots of numbers 👻

File: web/controllers/project_controller.ex [hunk 5]
```diff
@@ -50,7 +50,7 @@ defmodule CodeCorps.ProjectController do
 
     case Repo.insert(changeset) do
       {:ok, project} ->
-        project = Repo.preload(project, [:categories, :organization, :skills])
+        |> Repo.preload([:categories, :organization, :posts, :skills])
```
Comment (noslouch) [inline web/controllers/project_controller.ex]: this is dying on semaphore. I don't think the `case` clause heads pass a value to its body, so I think it needs to be something like...

``` elixir
project =
  project
  |> Repo.preload([:categories, :organization, :posts, :skills])
  ...
```

and then again the same thing around line 74

Comment (joshsmith) [reply web/controllers/project_controller.ex]: Yeah, this was dirty copy/paste in a merge. Will fix.

**Comment by begedin** [comment]
I did some digging on using `Repo.preload`, vs just `preload`

It looks like just `preload` is actually us using the weird linq/lambda style query synthax, so it works just off of a model module. `Repo.preload`, on the other hand is a method on the `Repo` module, which expects a struct or list of structs as the first argument, so it has to be called after `Repo.get*/Repo.all`. From what I can tell in the documentation and various other places, `Repo.preload` is the preferred approach and more commonly used. We should fallback to `preload` if we need some sort of query interpolation.

I corrected the controller so tests pass now. I also modified the index method a bit to simplify the query and possibly shorten it, making it more performant.

**Pull request merged**